java/annotations/src/main/java/org/opendcs/annotations/PropertySpec.java

```diff
@@ -37,4 +37,64 @@
     String value() default "";
     String propertySpecType() default "";
     String description() default "";
+    
+    /**
+     * Marks this property as required. 
+     * If true and no requirementGroups are specified, the property is placed in its own unique group.
+     * @return true if this property is required
+     */
+    boolean required() default false;
+    
+    /**
+     * Specifies the requirement groups this property belongs to.
+     * A property can belong to multiple requirement groups with different validation types.
+     * Empty array means the property is not part of any requirement group (optional).
+     * If required=true and requirementGroups is empty, a unique individual group is created.
+     *
+     * @return array of requirement group definitions
+     */
+    RequirementGroupDef[] requirementGroups() default {};
```

Comment:
I would name this either RequireGroup or RequirementGroupDefinition.

The annotations should probably get the most friendly name as that's what a downstream developer would interact with.